persist/fs/seek.go

```diff
@@ -181,6 +181,13 @@ func (s *seeker) Open(namespace ts.ID, shard uint32, blockStart time.Time) error
 		return err
 	}
 
+	if !s.keepUnreadBuf {
```

Comment:
I'm not following, I think I'm still resetting the buffer after the index is read? although the code is moved but it is literally still in the same place in the code path?